src/plugins/unified_search/public/dataview_picker/change_dataview.tsx

```diff
@@ -151,6 +152,10 @@ export function ChangeDataView({
     }
   }, [isTextBasedLangSelected, textBasedLanguage]);
 
+  const isAdHocSelected = useMemo(() => {
+    return adHocDataViews?.some((dataView) => dataView.title === trigger.title);
```

Comment:
The title is not necessarily unique (see how the selected library data view is shown with the icon because there's a matching ad hoc data view):
<img width="342" alt="Screenshot 2022-10-19 at 16 17 05" src="https://user-images.githubusercontent.com/1508364/196717011-17aeee30-14b1-43fb-ba8e-b65654e9f5a3.png">

Why not comparing the the id with `currentDataViewId` (didn't test)?

Reply:
You are right Joe! I forgot that when you dont give a name, the title will be the index pattern! Your suggestions works fine! Fixed!